plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/frontend.tsx

```diff
@@ -59,7 +87,13 @@ const productCollectionStore = {
 					'@wordpress/interactivity-router'
 				);
 
-				yield actions.navigate( ref.href );
+				// Todo: Remove after support for WP .6.6 is dropped.
+				if ( document.getElementById( 'wp-interactivity-data' ) ) {
+					const html = yield fetchUrlAndReplaceState( ref.href );
```

Comment:
Maybe we could implement a minimal cache to avoid fetching the HTML again before calling `navigate`. 🤔

Reply:
Another option could be calling `actions.prefetchOnHover` synchronously, i.e., without `yield`. That would fetch the HTML again, but it won't replace the cached one if it was already cached.

That would prevent waiting for the HTML again if it was already cached.

```suggestion
					actions.prefetchOnHover();
```

Reply:
To be honest, I wouldn't optimize anything since this will only be necessary during one version of WooCommerce, since when the next one after that comes out, it will no longer need to support WordPress 6.6.